src/internal/document_api/mod.rs

```diff
@@ -522,28 +708,148 @@ fn encrypt_document_core<'a, CR: rand::CryptoRng + rand::RngCore>(
+                encryption_errs: vec![encrypt_errs.into_iter().map(|e| e.into()).collect()]
+                    .into_iter()
+                    .concat(),
+            }
         })
     }
-    .into_future()
-    .and_then(move |(grants, encrypted_doc, other_errs)| {
-        document_create::document_create_request(auth, doc_id.clone(), document_name, grants)
-            .map(move |resp| (doc_id, resp, encrypted_doc, other_errs))
+}
+
+/// An encrypted document encryption key.
+///
+/// Once decrypted, the DEK serves as a symmetric encryption key.
+///
+/// It can also be useful to think of an EDEK as representing a "document access grant" to a user/group.
+#[derive(Debug, Clone, PartialEq)]
+pub struct EncryptedDek {
+    grant_to: WithKey<UserOrGroup>,
+    encrypted_dek_data: recrypt::api::EncryptedValue,
+}
+
+impl From<&EncryptedDek> for EncryptedDekP {
+    fn from(edek: &EncryptedDek) -> Self {
+        use crate::proto::edeks;
+        use recrypt::api as re;
+        let edek = edek.clone();
+
+        // transform the recrypt EncryptedValue to a edek proto
+        let mut proto_edek_data_no_pubk = match edek.encrypted_dek_data {
+            re::EncryptedValue::EncryptedOnceValue {
+                ephemeral_public_key,
+                encrypted_message,
+                auth_hash,
+                public_signing_key,
+                signature,
+            } => {
+                let mut proto_edek_data = EncryptedDekDataP::default();
+                {
+                    let mut proto_eph_pub_key = edeks::PublicKey::default();
+                    let (x, y) = ephemeral_public_key.bytes_x_y();
+                    proto_eph_pub_key.set_x(x[..].into());
+                    proto_eph_pub_key.set_y(y[..].into());
+                    proto_edek_data.set_ephemeralPublicKey(proto_eph_pub_key);
+                }
+                proto_edek_data.set_encryptedMessage(encrypted_message.bytes()[..].into());
+                proto_edek_data.set_signature(signature.bytes()[..].into());
+                proto_edek_data.set_authHash(auth_hash.bytes()[..].into());
+                proto_edek_data.set_publicSigningKey(public_signing_key.bytes()[..].into());
+
+                proto_edek_data
+            }
+            re::EncryptedValue::TransformedValue { .. } => {
+                unreachable!("Will be needed for decrypt!")
+            }
+        };
+
+        //convert the grants
+        let (proto_uog, proto_pub_key) = match edek.grant_to {
+            WithKey {
+                id:
+                    UserOrGroup::User {
+                        id: UserId(user_string),
+                    },
+                public_key,
+            } => {
+                let mut proto_uog = edeks::UserOrGroup::default();
+                proto_uog.set_userId(user_string.into());
+
+                let mut proto_pub_key = edeks::PublicKey::default();
+                proto_pub_key.set_x(public_key.as_bytes().into());
```

Comment:
Maintaining separate `x` and `y` is seeming less and less appealing. I think we've discussed treating the public key as a black box - any further thoughts?

Reply:
The structure of the proto is currently the exact same as recrypt. All consumers will have x and y because that's what recrypt provides (and requires). Seems like trying to put them together in the proto has limited gains.

Reply:
I thought the same thing. We keep pulling the x and y along for no reason. If want want to get rid of it, recrypt would be the place, however.

Reply:
Agree that it shouldn't stay split out in recrypt and get jammed together up higher. But should recrypt keep the internal details internal?

Reply:
I am very convinced that splitting the public key into x and y for public consumption was a mistake. We may have passed some point of no return as this design choice reverberates through many SDKs, REST endpoints, and serialization choices. We _could_ draw the line at edeks, combine these into on 64 byte blob and write some issues to clean that up in other places. IronOxide already supports serializing as a single array. Recrypt _could_ support in/out of a combined array and we could deprecate the other one.

We have to start somewhere... and it will never be easier than it is today to clean this up. @coltfred would you support making this change in the proto?

Reply:
We are punting this for now.